File: 1823_Find_the_Winner_of_the_Circular_Game/solution.c
int findTheWinner(int n, int k) {
    int *friends = (int*)malloc(n * sizeof(int));
    for (int i = 0; i < n; i++) friends[i] = i + 1;

    int len = n, index = 0;
    while (len > 1) {
        index = (index + k - 1) % len;         
        for (int i = index; i < len - 1; i++) friends[i] = friends[i + 1];
        len--;
    }

    int winner = friends[0];
    free(friends);
    return winner;
}

File: 1823_Find_the_Winner_of_the_Circular_Game/solution.py
class Solution:
    def findTheWinner(self, n: int, k: int) -> int:
        friends, index = list(range(1, n + 1)), 0
        while len(friends) > 1:
            index = (index + k - 1) % len(friends)
            friends.pop(index)
        return friends[0]
